src/tools/tool_cmdline.rs

```diff
@@ -94,18 +96,34 @@ pub fn cmdline_tool_from_yaml_value(
 pub struct CmdlineSession {
     cmdline_string: String,
     cmdline_workdir: String,
-    cmdline_process: tokio::process::Child,
+    cmdline_process: Box<dyn TokioChildWrapper>,
     #[allow(dead_code)]
     cmdline_stdout: BufReader<tokio::process::ChildStdout>,
     #[allow(dead_code)]
     cmdline_stderr: BufReader<tokio::process::ChildStderr>,
+    service_name: String,
 }
 
 impl IntegrationSession for CmdlineSession {
     fn as_any_mut(&mut self) -> &mut dyn Any {
         self
     }
     fn is_expired(&self) -> bool { false }
+    fn try_stop(&mut self) -> Box<dyn Future<Output = String> + Send + '_> {
+        Box::new(async {
+            info!("SERVICE STOP workdir {}:\n{:?}", self.cmdline_workdir, self.cmdline_string);
+            let t0 = tokio::time::Instant::now();
+            match Box::into_pin(self.cmdline_process.kill()).await {
```

Comment:
I'm not sure if .kill() to the process in Windows will terminate the job, to kill subprocesses, I think job.terminate needs to be called manuallly, not sure if there is a workaround on this?

Did you tested this with a process with subprocesses on windows? it may work with some strange underwork in windows but I'm not sure, I think this will not terminate the job.

Reply:
I tested it in linux and it kills subprocess and it's child (simple kill signal doesn't work)
docs says that JobObject wrapper should work as ProcessGroup on linux
if you have windows, I should ask for help with building and testing it

Reply:
I have Mac M1 so I can't install windows normally on it, I have it on a VM but it is barely usable, because I have only 8GB ram on my PC, last time I built refact-lsp it took 80 minutes, just crazy.

Reply:
But you are right about the doc, we should test in windows anyway, but probably it will work, it's even in examples I think